### PR TITLE
Connect: Remove leftovers from resource cache removal

### PR DESCRIPTION
**Pull request by ravicious**
* `GetAll*` RPCs were used in the resource cache and are no longer used.
* `ListApps` was pretty much never actually used as we didn't end up ever listing apps in Connect.
* `ClusterContext.state.searchValue` ceased to be used after we moved to fetching resources from the backend.

### CODE REVIEW

File: lib/teleterm/clusters/cluster_databases.go [hunk 1]
```diff
@@ -60,7 +60,9 @@ func (c *Cluster) GetDatabase(ctx context.Context, dbURI string) (*Database, err
 }
 
 // GetDatabases returns databases
-func (c *Cluster) GetAllDatabases(ctx context.Context) ([]Database, error) {
+// TODO(ravicious): Remove this method in favor of fetching a single database in GetDatabase.
+// https://github.com/gravitational/teleport/pull/14690#discussion_r927720600
+func (c *Cluster) getAllDatabases(ctx context.Context) ([]Database, error) {
```
Comment (ravicious) [inline lib/teleterm/clusters/cluster_databases.go:65]: I'd have removed it already but [I'm trying to establish what's the efficient way to do that](https://gravitational.slack.com/archives/C0DF0TPMY/p1678276982792309).

I'll probably submit another PR if I get to that.

File: web/packages/teleterm/src/services/tshd/fixtures/mocks.ts [hunk 2]
```diff
@@ -43,7 +42,6 @@ import {
 export class MockTshClient implements TshClient {
   listRootClusters: () => Promise<Cluster[]>;
   listLeafClusters: (clusterUri: string) => Promise<Cluster[]>;
-  listApps: (clusterUri: string) => Promise<Application[]>;
```
Comment (gzdunek) [inline web/packages/teleterm/src/services/tshd/fixtures/mocks.ts:46]: You can also remove `getAllServers`, `getAllDatabases`, `getAllServers` and `listServers`.

**Review by gzdunek — APPROVED** [review]
Thanks for this cleanup!

File: web/packages/teleterm/src/services/tshd/fixtures/mocks.ts (hunk 2 above)
Comment (ravicious) [reply web/packages/teleterm/src/services/tshd/fixtures/mocks.ts:46]: Thanks, too bad we cannot make TypeScript inform us about this here. There was this package for mocking classes automatically which was supposed to be added to jest directly but idk what happened with that.

**Review by espadolini — APPROVED** [review]

**Pull request merged**